In [1]:
pip install pandas 

Note: you may need to restart the kernel to use updated packages.



[notice] A new release of pip is available: 23.2.1 -> 24.0
[notice] To update, run: python.exe -m pip install --upgrade pip


In [ ]:
notas_funcionario1 = [8,7,6,-2,9]
notas_funcionario2 = [6,5,7,8,7]



In [24]:
from itertools import groupby
import pandas as pd

pecas1={'nome': 'Camiseta', 'categoria': 'Calçados', 'preço': 29.99}
pecas2={'nome': 'Calça', 'categoria': 'Roupas', 'preço': 49.99}
pecas3={'nome': 'Tênis', 'categoria': 'Calçados', 'preço': 99.99}
pecas4={'nome': 'Meia', 'categoria': 'Acessórios', 'preço': 9.99}
pecas5={'nome': 'Boné', 'categoria': 'Acessórios', 'preço': 19.99}

produtos = [pecas1,pecas2,pecas3,pecas4,pecas5]

juncao = lambda x: x["categoria"]    

resultado = groupby(sorted(produtos, key=juncao), juncao)

for key, group in resultado:
    print(f"Categoria: {key}")
    for value in group:
        print(f"{value}")  

media_roupas = []
media_calçados = []
media_acessórios = []

for i in range(len(produtos)):
    if produtos[i]["categoria"] == "Roupas":
        media_roupas.append(produtos[i]["preço"])
    elif produtos[i]["categoria"] == "Calçados":
        media_calçados.append(produtos[i]["preço"])  
    elif produtos[i]["categoria"] == "Acessórios":
        media_acessórios.append(produtos[i]["preço"])  


df = pd.DataFrame(produtos)
media_categoria = df.groupby('categoria')['preço'].mean()

print(media_categoria)

Categoria: Acessórios
{'nome': 'Meia', 'categoria': 'Acessórios', 'preço': 9.99}
{'nome': 'Boné', 'categoria': 'Acessórios', 'preço': 19.99}
Categoria: Calçados
{'nome': 'Camiseta', 'categoria': 'Calçados', 'preço': 29.99}
{'nome': 'Tênis', 'categoria': 'Calçados', 'preço': 99.99}
Categoria: Roupas
{'nome': 'Calça', 'categoria': 'Roupas', 'preço': 49.99}
categoria
Acessórios    14.99
Calçados      64.99
Roupas        49.99
Name: preço, dtype: float64


Categoria: Acessórios
{'nome': 'Meia', 'categoria': 'Acessórios', 'preço': 9.99}
{'nome': 'Boné', 'categoria': 'Acessórios', 'preço': 19.99}
Categoria: Calçados
{'nome': 'Camiseta', 'categoria': 'Calçados', 'preço': 29.99}
{'nome': 'Tênis', 'categoria': 'Calçados', 'preço': 99.99}
Categoria: Roupas
{'nome': 'Calça', 'categoria': 'Roupas', 'preço': 49.99}
Média de preços das roupas: 49.99
Média de preços das calçados: 64.99
Média de preços das acessórios: 14.989999999999998

In [25]:
def simular_estoque(vendas, reposicoes, estoque_inicial):
    estoque_real = estoque_inicial
    for dia, venda, reposicao in zip(range(1, len(vendas) + 1), vendas, reposicoes):
        estoque_real -= venda
        estoque_real += reposicao
        print(f"No dia {dia}:, Vendas: {venda}, Reposições: {reposicao}, Estoque atual: {estoque_real}")
    return estoque_real

vendas = [10,15,8,12]  
reposicoes = [5,7,6,5]  
estoque_inicial = 100  
estoque_final = simular_estoque(vendas, reposicoes, estoque_inicial)
print(f"Estoque final: {estoque_final}")

No dia 1:, Vendas: 10, Reposições: 5, Estoque atual: 95
No dia 2:, Vendas: 15, Reposições: 7, Estoque atual: 87
No dia 3:, Vendas: 8, Reposições: 6, Estoque atual: 85
No dia 4:, Vendas: 12, Reposições: 5, Estoque atual: 78
Estoque final: 78


In [18]:
import pandas as pd

produtos = [
    {'nome': 'Camiseta', 'categoria': 'Roupas', 'preco': 29.99},
    {'nome': 'Calça', 'categoria': 'Roupas', 'preco': 49.99},
    {'nome': 'Tênis', 'categoria': 'Calçados', 'preco': 99.99},
    {'nome': 'Meia', 'categoria': 'Acessórios', 'preco': 9.99},
    {'nome': "Boné", "categoria": "Acessórios", "preco": 19.99}
]

df = pd.DataFrame(produtos)
media_por_categoria = df.groupby('categoria')['preco'].mean()

print(media_por_categoria)


categoria
Acessórios    14.99
Calçados      99.99
Roupas        39.99
Name: preco, dtype: float64


In [9]:

notas_funcionario1 = [8, 7, 6, -2, 9]
notas_funcionario2 = [6, 5.5, 7, 8.3, 7]

df = {
    'a': [8, 7, 6, 2, 9],
    'b':[6, 5.5, 7, 8.3, 7]
}

df = pd.DataFrame(df)
df

a    b
0  8  6.0
1  7  5.5
2  6  7.0
3  2  8.3
4  9  7.0

In [8]:
import pandas as pd 


def calcular_media(notas): 
    notas_validas = list(filter(lambda nota: nota >= 0, notas))
    media = sum(notas_validas) / len(notas_validas) if notas_validas else 0
    return media

media1 = calcular_media(notas_funcionario1)
media2 = calcular_media(notas_funcionario2)

print(f'Média do funcionário 1: {media1}')
print(f'Média do funcionário 2: {media2}')


Média do funcionário 1: 6.4
Média do funcionário 2: 6.76


In [11]:
notas_funcionario1 = [8,7,6,-2,9]
notas_funcionario2 = [6,5,7,8,7]
def calcular_media_funcionario():
    notas_validas1 = list(filter(lambda x:( x > 0),notas_funcionario1)) # no enunciado ele pede para filtrar a nota negativa, por isso eu usei o x>0, pq toda nota menor que 0 ele desconsidera 
    notas_validas2 = list(filter(lambda x:(x > 0),notas_funcionario2))
    media_funcionario1 = sum(notas_validas1)/len(notas_validas1)
    media_funcionario2 = sum(notas_validas2)/len(notas_validas2)
    print("Média do funcionário 1: ", media_funcionario1)
    print("Média do funcionário 2: ", media_funcionario2)
 
calcular_media_funcionario()

Média do funcionário 1:  7.5
Média do funcionário 2:  6.6


In [10]:
#3 
import pandas as pd 
df = pd.DataFrame({
    'Texto':['O texto está agradável para passear e curtir',
            'O parque é um ótimo lugar para curtir',
            'As pessoas gostam de passear no parque com o tempo agradável']
})

def soma(dftexto):
    dftexto['Contagem'] = dftexto['Texto'].apply(lambda texto:len(texto))
    return dftexto

soma(df)


Texto  Contagem
0       O texto está agradável para passear e curtir        44
1              O parque é um ótimo lugar para curtir        37
2  As pessoas gostam de passear no parque com o t...        60

In [12]:
# 5
contas = {'Salarios':[5000, 5500, 6000, 5800],
            'Aluguel': [1500, 1500, 1600, 1600],
            'Fornecedores': [2000, 2200, 2300, 2100]}

df_contas = pd.DataFrame(contas)

media_salarios = df_contas['Salarios'].apply(lambda x: x).mean()
media_aluguel = df_contas['Aluguel'].apply(lambda x: x).mean()
media_fornecedores = df_contas['Fornecedores'].apply(lambda x: x).mean()

print(f'Média Salarios:{media_salarios} . Média alguel {media_aluguel}. Média fornecedores {media_fornecedores}.')

Média Salarios:5575.0 . Média alguel 1550.0. Média fornecedores 2150.0.


In [9]:
# 6
import pandas as pd 
def calcular_raizes_quadradicas(a,b,c):
    discriminante = b**2 - 4 * a * c
    discr_calcu = discriminante

    calcular_x1_x2 = lambda a, b, c: (
        (-b + discr_calcu ** 0.5) / (2*a),
        (-b - discr_calcu ** 0.5) / (2*a)
    )

    return calcular_x1_x2(a,b,c)

raizes = calcular_raizes_quadradicas(1, -3, 2)
print("Raiz x1:", raizes[0])
print("Raiz x2:", raizes[1])

Raiz x1: 2.0
Raiz x2: 1.0


In [6]:
#6
def calular_raizes_quadraticas(a,b,c):
    discriminante = b**2 - 4*a*c

    calcular_x1_x2 = lambda a, b, c: (
        (-b + discriminante**0,5) / (2*a),
        (-b - discriminante**0,5) / (2*a)
    )
    return calcular_x1_x2(a,b,c)

raizes = calular_raizes_quadraticas(1,-3,2)
print('Raiz x1:', raizes[0])
print('Raiz x2:', raizes[1])

TypeError: unsupported operand type(s) for /: 'tuple' and 'int'

In [5]:
# 4 
def simular_estoque(vendas, estoque_inicial, reposicoes):
    estoque = estoque_inicial.copy()
    for i in range(len(vendas)):
        estoque[i] -= vendas[i]  # Subtrai as vendas do estoque
        estoque[i] += reposicoes[i]  # Adiciona as reposições ao estoque
        # Aqui você pode adicionar uma função definida pelo usuário para calcular o novo nível de estoque
        # Por exemplo: estoque[i] = calcular_novo_estoque(estoque[i])
    return estoque

# Exemplo de uso da função:
vendas = [10, 15, 8, 12]
estoque_inicial = [100]
reposicoes = [5, 7, 6, 5]

estoque_final = simular_estoque(vendas, estoque_inicial, reposicoes)
print(estoque_final)


IndexError: list index out of range

In [4]:
def simular_estoque(vendas, reposicoes, estoque_inicial):
    estoque_real = estoque_inicial
    for dia, venda, reposicao in zip(range(1, len(vendas) + 1), vendas, reposicoes):
        estoque_real -= venda
        estoque_real += reposicao
        print(f"No dia {dia}:, Vendas: {venda}, Reposições: {reposicao}, Estoque atual: {estoque_real}")
    return estoque_real

vendas = [10,15, 8, 12]  
reposicoes = [5,7, 6, 5]  
estoque_inicial = 100  
estoque_final = simular_estoque(vendas, reposicoes, estoque_inicial)
print(f"Estoque final: {estoque_final}")

No dia 1:, Vendas: 10, Reposições: 5, Estoque atual: 95
No dia 2:, Vendas: 15, Reposições: 7, Estoque atual: 87
No dia 3:, Vendas: 8, Reposições: 6, Estoque atual: 85
No dia 4:, Vendas: 12, Reposições: 5, Estoque atual: 78
Estoque final: 78


In [33]:
from itertools import groupby
 
produto1={'nome': 'Camiseta', 'categoria': 'Roupas', 'preço': 29.99}
produto2={'nome': 'Calça', 'categoria': 'Roupas', 'preço': 49.99}
produto3={'nome': 'Tênis', 'categoria': 'Calçados', 'preço': 99.99}
produto4={'nome': 'Meia', 'categoria': 'Acessórios', 'preço': 9.99}
produto5={'nome': 'Boné', 'categoria': 'Acessórios', 'preço': 19.99}
 
produtos = [produto1,produto2,produto3,produto4,produto5]
 
agrupar = lambda x: x["categoria"]    
 
resultado = groupby(sorted(produtos, key=agrupar), agrupar)
 
 
for key, group in resultado:
    print(f"Chave: {key}")
    for value in group:
        print(f"- {value}") 
        
  

Chave: Acessórios
- {'nome': 'Meia', 'categoria': 'Acessórios', 'preço': 9.99}
- {'nome': 'Boné', 'categoria': 'Acessórios', 'preço': 19.99}
Chave: Calçados
- {'nome': 'Tênis', 'categoria': 'Calçados', 'preço': 99.99}
Chave: Roupas
- {'nome': 'Camiseta', 'categoria': 'Roupas', 'preço': 29.99}
- {'nome': 'Calça', 'categoria': 'Roupas', 'preço': 49.99}


In [ ]:
#7
a) correta
b) correta
c) Incorreto
d) Incorreto
e) Correto 